orlo/user_auth.py

```diff
@@ -76,63 +57,103 @@ def verify_password(self, password):
         return rc
 
     def generate_auth_token(self, expiration=3600):
-        s = Serializer(app.config['SECRET_KEY'], expires_in=expiration)
-        return s.dumps({'id': self.name})
+        secret_key = config.get('security', 'secret_key')
+        s = Serializer(secret_key, expires_in=expiration)
+        return s.dumps({'id': self.username})
 
     @staticmethod
     def verify_auth_token(token):
-        s = Serializer(app.config['SECRET_KEY'])
+        app.logger.debug("Verify auth token called")
+        s = Serializer(config.get('security', 'secret_key'))
         try:
             data = s.loads(token)
         except SignatureExpired:
             return None  # valid token, but expired
         except BadSignature:
             return None  # invalid token
-        name = data['id']
-        return User(name)
+        user = data['id']
+        return user
 
 
 @user_auth.verify_password
-def verify_password(username=None, password=None):
-    app.logger.debug("Verify_password called")
-
-    if not password:
-        return False
-
-    user = User(username)
-    if not user.verify_password(password):
-        return False
-
-    set_current_user_as(user)
-    return True
+def verify_password(username_or_token=None, password=None):
```

Comment:
I really don't like the idea that a user name could be a token, but we can fix this up later